File: 2368-reachable-nodes-with-restrictions/2368-reachable-nodes-with-restrictions.cpp
class Solution {
public:
    vector<int>dp[100006];
    int vis[100006];
    void find(int node,unordered_set<int>&st)
    {
        vis[node]=1;
        for(auto &it:dp[node])
        {
            if(!vis[it]&&st.find(it)==st.end())
            {
                find(it,st);
            }
        }
    }
    int reachableNodes(int n, vector<vector<int>>& edges, vector<int>& res) 
    {
        unordered_set<int>st(res.begin(),res.end());
        for(auto &it:edges)
        {
            dp[it[0]].push_back(it[1]);
            dp[it[1]].push_back(it[0]);
        }
        find(0,st);
        return accumulate(vis,vis+n,0);
    }
};

File: 2368-reachable-nodes-with-restrictions/2368-reachable-nodes-with-restrictions.py
class Solution:
    def reachableNodes(self, n: int, edges: List[List[int]], res: List[int]) -> int:
        dp = defaultdict(set)
        for a,b in edges:
            dp[a].add(b)
            dp[b].add(a)
        vis=set(res)    
        vis.add(0)
        q=deque([0])
        count=0
        while q:
            temp=q.popleft()
            for it in dp[temp]:
                if it not in vis:
                    # count+=1
                    q.append(it)
                    vis.add(it)
        return len(vis)-len(res)
        